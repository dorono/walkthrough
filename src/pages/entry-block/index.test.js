import React from 'react';
import {shallow} from 'enzyme';
import {EntryBlockPage} from './index';

const defaultProps = {
    data: {
        chain: {},
        next: {},
        prev: {},
    },
    location: {},
};

jest.mock('../../utils/date', () => {
    return {
        currentTimezone: () => '11/11/11',
        formatDateLong: () => '11111111',
    };
});

describe('EntryBlock', () => {
    it('should render without errors', () => {
        shallow(<EntryBlockPage {...defaultProps} />);
    });

    it('should match snapshot', () => {
        expect(shallow(<EntryBlockPage {...defaultProps} />)).toMatchSnapshot();
    });
});
