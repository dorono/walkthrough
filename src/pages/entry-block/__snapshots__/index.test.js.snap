// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryBlock should match snapshot 1`] = `
<div>
  <Container
    primary={true}
    showFullWidthBanner={false}
    title="Entry block"
  >
    <Vertical>
      <Box
        type="fill"
      >
        <Label>
          NEXT ENTRY BLOCK
        </Label>
        <withRouter(Hash)
          type="eblock"
        />
      </Box>
      <Box
        type="outline"
      >
        <Label>
          KEYMR
        </Label>
        <withRouter(Hash)
          type="eblock"
        />
      </Box>
      <Box
        type="fill"
      >
        <Label>
          PREVIOUS ENTRY BLOCK
        </Label>
        <withRouter(Hash)
          type="eblock"
        />
      </Box>
      <Box
        type="fill"
      >
        <Label>
          CHAIN
        </Label>
        <withRouter(Hash)
          type="chain"
        />
      </Box>
      <Box
        type="fill"
      >
        <Label>
          PARENT DIRECTORY BLOCK
        </Label>
        <DirectoryBlockLink
          isLink={true}
          type="dblock"
        />
      </Box>
      <Box>
        <Label>
          STARTED (
          11/11/11
          )
        </Label>
        11111111
      </Box>
    </Vertical>
  </Container>
  <DataLoader
    entriesUrl="/eblocks/undefined/entries"
    renderContent={[Function]}
  />
</div>
`;
